src/sentry/api/serializers/models/group.py

```diff
@@ -416,6 +417,13 @@ def get_attrs(self, item_list, user):
 
         snuba_stats = self._get_group_snuba_stats(item_list, seen_stats)
 
+        sample_groups = tagstore.get_group_list_tag_value(
```

Comment:
Adding and additional snuba query to every group serialize operation for sample groups seems expensive for the amount of benefit we are going to provide. Should we try writing down that an issue is a sample issue somewhere that is more efficient to read from? Perhaps something like `GroupMeta` could work?

Reply:
@Neo-Zhixing I agree with Mark, I don't want to add a new Snuba query for this. Maybe I'm not understanding why we need this, is it because we don't have the tags available in the serializer without this query?

Reply:
We don't have tags available in the listing page because we don't query for them - so yeah. Tags are associated with `events` while this page query for `Groups`. 

If we do save the sample tag in GroupMeta, we'd be saving it in two different places, and I'm not sure how we could modify GroupMeta when creating an event.

Reply:
>  I'm not sure how we could modify GroupMeta when creating an event.

Event processing plugins can do this. We have a few internal plugins that do this by implementing the `post_process()` plugin hook. An example of this is the old Jira plugin.